### PR TITLE
[CHK-1076] feat(domain): add `TransactionExpired` aggregate

### PR DESCRIPTION
**Pull request by giovanniberti**
#### List of Changes

* add `TransactionExpired` aggregate

#### Motivation and Context

This PR adds a `TransactionExpired` aggregate. This aggregate is used to handle transactions in an `EXPIRED` and to correctly construct aggregates though `EXPIRED` events.

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by giovanniberti** [comment]
**NOTE ON CODE COVERAGE TO REVIEWERS:**

* `TransactionWithCompletedAuthorization` is a legacy class which should be deprecated (in fact right now it's not used anywhere and can be safely deleted)
* Same thing for `TransactionWithActivationRequested`
* Other stuff is missing branch coverage from Lombok-generated code such as `@Equals`

File: src/main/java/it/pagopa/ecommerce/commons/domain/TransactionExpired.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            TransactionExpiredEvent event
+    ) {
+        super(baseTransaction, event.getData());
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public Transaction applyEvent(Object event) {
+        return this;
+    }
```
Comment (pietro-tota) [inline src/main/java/it/pagopa/ecommerce/commons/domain/TransactionExpired.java:38]: Are we missing here the transition to TransactionRefunded? TransactionRefunded aggregate is not present as for now and maybe is better to add it into a separate PR, what do you think?

**Review by pietro-tota — APPROVED** [review]
🚀

File: src/main/java/it/pagopa/ecommerce/commons/domain/TransactionExpired.java (hunk 1 above)
Comment (giovanniberti) [reply src/main/java/it/pagopa/ecommerce/commons/domain/TransactionExpired.java:38]: Correct, and that's _exactly_ what I was thinking 😉

**Pull request merged**